File: backend_dotnet/fruit_api/fruit_api/Models/Inventory.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("Inventory")]
public class Inventory
{
    [Key]
    [Column("inventoryId")]
    [StringLength(20)]
    public string InventoryId { get; set; } = null!;

    [Required]
    [Column("productId")]
    [StringLength(20)]
    public string ProductId { get; set; } = null!;

    [Required]
    [MaxLength(50)]
    [Column("batchCode")]
    public string BatchCode { get; set; } = string.Empty;

    [Required]
    [Column("quantity")]
    public int Quantity { get; set; }

    [Required]
    [Column("manufactureDate", TypeName = "DATE")]
    public DateTime? ManufactureDate { get; set; }

    [Required]
    [Column("expiryDate", TypeName = "DATE")]
    public DateTime ExpiryDate { get; set; }

    [Required]
    [Column("importPrice", TypeName = "decimal(12,2)")]
    public decimal? ImportPrice { get; set; }

    [Column("importDate")]
    public DateTime ImportDate { get; set; } = DateTime.Now;

    [Column("status")]
    [StringLength(20)]
    public string Status { get; set; } = "in_stock";

    [Column("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    // Navigation properties
    [ForeignKey("ProductId")]
    public virtual Product? Product { get; set; }

    // Helper properties (không lưu trong database)
    [NotMapped]
    public bool IsExpired => ExpiryDate < DateTime.Today;

    [NotMapped]
    public bool IsInStock => Status == "in_stock" && Quantity > 0 && !IsExpired;

    [NotMapped]
    public int DaysToExpiry => ExpiryDate > DateTime.Today
        ? (ExpiryDate - DateTime.Today).Days
        : 0;
}

File: frontend_flutter/web_admin - Copy/lib/models/inventory.dart
// lib/models/inventory.dart

class Inventory {
  final String inventoryId;
  final String productId;
  final String productName;
  final String? productImage;
  final String batchCode;
  final int quantity;
  final DateTime? manufactureDate;
  final DateTime expiryDate;
  final double? importPrice;
  final DateTime importDate;
  final String status;
  final String statusText;
  final bool isExpired;
  final int daysToExpiry;
  final DateTime createdAt;

  Inventory({
    required this.inventoryId,
    required this.productId,
    required this.productName,
    this.productImage,
    required this.batchCode,
    required this.quantity,
    this.manufactureDate,
    required this.expiryDate,
    this.importPrice,
    required this.importDate,
    required this.status,
    required this.statusText,
    required this.isExpired,
    required this.daysToExpiry,
    required this.createdAt,
  });

  factory Inventory.fromJson(Map<String, dynamic> json) {
    return Inventory(
      inventoryId: json['inventoryId']?.toString() ?? '',
      productId: json['productId']?.toString() ?? '',
      productName: json['productName'] ?? '',
      productImage: json['productImage'],
      batchCode: json['batchCode'] ?? '',
      quantity: json['quantity'] ?? 0,
      manufactureDate: json['manufactureDate'] != null 
          ? DateTime.parse(json['manufactureDate']) 
          : null,
      expiryDate: json['expiryDate'] != null 
          ? DateTime.parse(json['expiryDate']) 
          : DateTime.now(),
      importPrice: json['importPrice']?.toDouble(),
      importDate: json['importDate'] != null 
          ? DateTime.parse(json['importDate']) 
          : DateTime.now(),
      status: json['status'] ?? 'in_stock',
      statusText: json['statusText'] ?? 'Còn hàng',
      isExpired: json['isExpired'] ?? false,
      daysToExpiry: json['daysToExpiry'] ?? 0,
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
    );
  }

  // Helper properties
  bool get isInStock => status == 'in_stock' && !isExpired && quantity > 0;
  bool get isExpiringSoon => daysToExpiry > 0 && daysToExpiry <= 7;
  String get expiryStatus {
    if (isExpired) return 'Hết hạn';
    if (isExpiringSoon) return 'Sắp hết hạn';
    return 'Còn hạn';
  }
}

// DTO cho danh sách lô hàng (kiểu OrderListDto)
class InventoryListDto {
  final String inventoryId;
  final String productId;
  final String productName;
  final String? productImage;
  final String batchCode;
  final int quantity;
  final DateTime? manufactureDate;
  final DateTime expiryDate;
  final double? importPrice;
  final DateTime importDate;
  final String status;
  final String statusText;
  final bool isExpired;
  final int daysToExpiry;

  InventoryListDto({
    required this.inventoryId,
    required this.productId,
    required this.productName,
    this.productImage,
    required this.batchCode,
    required this.quantity,
    this.manufactureDate,
    required this.expiryDate,
    this.importPrice,
    required this.importDate,
    required this.status,
    required this.statusText,
    required this.isExpired,
    required this.daysToExpiry,
  });

  factory InventoryListDto.fromJson(Map<String, dynamic> json) {
    return InventoryListDto(
      inventoryId: json['inventoryId']?.toString() ?? '',
      productId: json['productId']?.toString() ?? '',
      productName: json['productName'] ?? '',
      productImage: json['productImage'],
      batchCode: json['batchCode'] ?? '',
      quantity: json['quantity'] ?? 0,
      manufactureDate: json['manufactureDate'] != null 
          ? DateTime.parse(json['manufactureDate']) 
          : null,
      expiryDate: json['expiryDate'] != null 
          ? DateTime.parse(json['expiryDate']) 
          : DateTime.now(),
      importPrice: json['importPrice']?.toDouble(),
      importDate: json['importDate'] != null 
          ? DateTime.parse(json['importDate']) 
          : DateTime.now(),
      status: json['status'] ?? 'in_stock',
      statusText: json['statusText'] ?? 'Còn hàng',
      isExpired: json['isExpired'] ?? false,
      daysToExpiry: json['daysToExpiry'] ?? 0,
    );
  }
}